### PR TITLE
Soolaf , homework week4

### PR DESCRIPTION
**Pull request by soolaf**

### CODE REVIEW

**Comment by SarKurd** [comment]
Sorry your app doesn't work.

Here the solution.

Any component that you use from `react-router-dom` should be inside `BrowserRouter` or `HashRouter`(these 2 are most popular)

**First way**: surround whole `app.js return `with browserRouter`
```javascript
import React, { Component } from 'react';
import { BrowserRouter, Route, Link, Switch } from 'react-router-dom';

import StaticList from './Components/Week1/App';
import DynamicList from './Components/Week2/App';
import APICall from './Components/Week3/App';
import './App.css';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div className="App">

          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/week1">Week1</Link>
            </li>
            <li>
              <Link to="/week2">Week2</Link>
            </li>
            <li>
              <Link to="/week3">Week3</Link>
            </li>
          </ul>

          <div>

            <Switch>
              <Route exact path="/" render={() => <h3> Homeworks </h3>} />
              <Route path="/week1" component={StaticList} />
              <Route exact path="/week2" component={DynamicList} />
              <Route path="/week3" component={APICall} />
            </Switch>

          </div>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;

```

**Second way** which i prefer is to put browser router in `Index.js`
```javascript
import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';
import * as serviceWorker from './serviceWorker';
import { BrowserRouter } from 'react-router-dom';


ReactDOM.render(
    < BrowserRouter >
        <App />
    </BrowserRouter >
    , document.getElementById('root'));

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();

```

File: homework-submission/myapp/src/App.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import DynamicList from './Components/Week2/App';
+import APICall from './Components/Week3/App';
+import './App.css';
+
+class App extends Component {
+  render() {
+    return (
+      <div className="App">
+        <MemoryRouter>
+          <ul>
```
Comment (SarKurd) [inline homework-submission/myapp/src/App.js]: Why are you using `MemoryRouter`?!!

File: homework-submission/myapp/src/Components/Week1/App.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import './Style.css';
+
+class List extends Component {
+  render() {
+    const description = this.props.description;
+    const deadline = this.props.deadline;
```
Comment (SarKurd) [inline homework-submission/myapp/src/Components/Week1/App.js]: It's a good practice to use no more than one component inside each file

File: homework-submission/myapp/src/Components/Week3/Components/Index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        this.setState(
+          {
+            isLoaded: true,
+            items,
+          },
+          () => console.log(items),
+        ),
+      )
+      .catch(error => console.log(error));
+  }
```
Comment (SarKurd) [inline homework-submission/myapp/src/Components/Week3/Components/Index.js]: Update the state for error case too

File: homework-submission/myapp/src/App.js (hunk 1 above)
Comment (soolaf) [reply homework-submission/myapp/src/App.js]: if I did not put it I git this error(Error: Invariant failed: You should not use <Link> outside a <Router>)

Comment (SarKurd) [reply homework-submission/myapp/src/App.js]: As i said in the comment BrowserRouter should wrap your whole app which i gave you two solutions

Link is a component which basically contains `a` tag

File: homework-submission/myapp/src/Components/Week3/Components/Index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        this.setState(
+          {
+            error: handleErrors,
+            isLoaded: true,
+            items,
+          },
+          () => console.log(items),
+        ),
+      )
+      .catch(error => this.setState({ error, isLoading: false }));
```
Comment (SarKurd) [inline homework-submission/myapp/src/Components/Week3/Components/Index.js]: Also you have to change error to true or a message so we know something is went wrong

Also in render make a condition if error !== null then return error message

File: homework-submission/myapp/src/Components/Week3/Components/Index.js (hunk 3 above)
Comment (soolaf) [reply homework-submission/myapp/src/Components/Week3/Components/Index.js]: Thank  you very much it's work

**Pull request closed without merging**